src/hooks/useBlockTimestamp.ts

```diff
@@ -1,20 +1,21 @@
-import { useState, useEffect } from 'react'
+import { useAtomValue } from 'jotai'
+import { useEffect, useState } from 'react'
+import { chainIdAtom } from 'state/atoms'
+import { wagmiConfig } from 'state/chain'
 import { useBlockNumber } from 'wagmi'
-import { usePublicClient } from 'wagmi'
-import useRToken from './useRToken'
-import { ChainId } from 'utils/chains'
+import { getBlock } from 'wagmi/actions'
 
 function useBlockTimestamp(blockNumber?: number) {
-  const rToken = useRToken()
+  const chainId = useAtomValue(chainIdAtom)
```

Comment:
We were using the `chainId` of the RToken before, is this change ok?

Reply:
chainIdAtom always tracks RToken chainId